miner/proposal_builder.go

```diff
@@ -668,17 +680,17 @@ func (pb *ProposalBuilder) build(ctx context.Context, lid types.LayerID) error {
 			)
 			if err := pb.publisher.Publish(ctx, pubsub.ProposalProtocol, codec.MustEncode(proposal)); err != nil {
 				ss.log.Error("failed to publish proposal",
-					log.Context(ctx),
-					log.Uint32("lid", proposal.Layer.Uint32()),
-					log.Stringer("id", proposal.ID()),
-					log.Err(err),
+					log.ZContext(ctx),
+					zap.Uint32("lid", proposal.Layer.Uint32()),
```

Comment:
How about a little helper to make life easier and logs consistent (using same key for layer in logs):
```go
func (l LayerID) MarshalLogObject(enc zapcore.ObjectEncoder) error {
	enc.AddUint32("layer", l.Uint32())
	return nil
}
```
```suggestion
					zap.Inline(proposal.Layer),
```

Reply:
I'm not sure if this is a good idea. We had issues before where 2 layers or epochs were added to a log line and both were serialized with the same key, which in the case of zap luckily doesn't replace the first with the second but leads to invalid json that might not be able to be parsed by log aggregators.

Having to provide the key explicitly makes the issue more visible when it happens.

Reply:
On the other hand it makes searching through the logs more difficult because same things are logged with different keys, like: 
- layer_id
- layer
- lid